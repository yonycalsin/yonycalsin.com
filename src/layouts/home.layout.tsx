import * as React from 'react'

import { FloatSocials, Navbar } from '~/components'
import NightModeButton from '~/components/night-mode-button'

type HomeLayoutProps = {
  children: React.ReactNode
  isStandaloneBrand?: boolean
}

export const HomeLayout = (props: HomeLayoutProps) => {
  const { children } = props

  return (
    <>
      <NightModeButton />
      <Navbar />

      <div className="container">{children}</div>

      <footer className="bg-gray-50 dark:bg-gray-900">
        <div className="container border-t-2 border-gray-100 dark:border-gray-700">
          <div className="flex flex-col md:flex-row flex-wrap justify-between items-center ">
            <p>
              <small>
                Created By{' '}
                <i>
                  <b>Yony Calsin</b>
                </i>{' '}
              </small>
            </p>

            <p>
              <small>With NextJs, Typescript, TailwindCss, and Vercel</small>
            </p>
          </div>
        </div>
      </footer>

      <FloatSocials />
    </>
  )
}
