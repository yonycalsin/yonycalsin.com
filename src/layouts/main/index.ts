export * from './components'

// @yonycalsin/eslint-import-sort/typescript-react-exports group
export { default as MainLayout } from './main'
