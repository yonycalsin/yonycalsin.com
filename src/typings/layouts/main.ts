import type * as React from 'react'

interface MainLayoutProps {
  children: React.ReactNode
}

export type { MainLayoutProps }
