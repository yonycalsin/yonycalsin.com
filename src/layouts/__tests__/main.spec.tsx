import { render } from '@testing-library/react'

import { overrideFeatures, TEST_ENVS } from 'tests'
import { MainLayout } from 'layouts'

describe('MainLayout', () => {
  beforeAll(() => {
    process.env = overrideFeatures()
  })

  afterAll(() => {
    process.env = TEST_ENVS
  })

  it('renders the main layout', () => {
    const view = render(
      <MainLayout>
        <h1>Content</h1>
      </MainLayout>,
    )

    expect(view.baseElement).toMatchSnapshot()
  })
})
