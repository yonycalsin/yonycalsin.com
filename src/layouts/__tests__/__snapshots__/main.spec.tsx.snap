// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout renders the main layout 1`] = `
<body>
  <div>
    <button
      class="fixed left-4 lg:left-8 top-20 lg:top-8 z-50 text-2xl lg:text-4xl"
      role="button"
    >
      <svg
        class="hidden dark:block"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.473 11a4.5 4.5 0 0 0-8.72-.99A3 3 0 0 0 3 16h8.5a2.5 2.5 0 0 0 0-5h-.027z"
        />
        <path
          d="M10.5 1.5a.5.5 0 0 0-1 0v1a.5.5 0 0 0 1 0v-1zm3.743 1.964a.5.5 0 1 0-.707-.707l-.708.707a.5.5 0 0 0 .708.708l.707-.708zm-7.779-.707a.5.5 0 0 0-.707.707l.707.708a.5.5 0 1 0 .708-.708l-.708-.707zm1.734 3.374a2 2 0 1 1 3.296 2.198c.199.281.372.582.516.898a3 3 0 1 0-4.84-3.225c.352.011.696.055 1.028.129zm4.484 4.074c.6.215 1.125.59 1.522 1.072a.5.5 0 0 0 .039-.742l-.707-.707a.5.5 0 0 0-.854.377zM14.5 6.5a.5.5 0 0 0 0 1h1a.5.5 0 0 0 0-1h-1z"
        />
      </svg>
      <svg
        class="block dark:hidden"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"
        />
        <path
          d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"
        />
      </svg>
    </button>
    <div
      class="w-full z-10 shadow-sm"
    >
      <div
        class="container py-5 md:py-6 flex justify-between items-center"
      >
        <a
          href="/"
        >
          <img
            alt="Logo"
            data-nimg="1"
            decoding="async"
            height="32"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
            style="color: transparent;"
            width="32"
          />
        </a>
        <div
          class="space-x-4"
        >
          <a
            class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
            href="/projects"
          >
            Projects
          </a>
          <a
            class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
            href="/me"
          >
            About Me
          </a>
        </div>
      </div>
    </div>
    <h1>
      Content
    </h1>
    <footer
      class="my-6"
    >
      <div
        class="container"
      >
        <footer
          class="p-4 bg-white dark:bg-gray-800 rounded-lg shadow flex flex-col md:flex-row md:items-center md:justify-between md:p-6"
        >
          <span
            class="text-gray-500"
          >
            © 2022
             
            <a
              class="hover:underline"
              href="https://yonycalsin.com"
            >
              Yony Calsin™
            </a>
            . All Rights Reserved.
          </span>
          <ul
            class="flex flex-wrap gap-3 items-center justify-center"
          >
            <li>
              <a
                class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
                href="/me"
              >
                About
              </a>
            </li>
            <li>
              <a
                class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
                href="/contact"
              >
                Contact
              </a>
            </li>
          </ul>
        </footer>
      </div>
    </footer>
    <div
      class="space-y-5 fixed bottom-8 left-8 hidden lg:flex flex-col"
    >
      <a
        class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
        href="https://github.com/yonycalsin"
        rel="noreferrer"
        target="_blank"
      >
        Github
      </a>
      <a
        class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
        href="https://www.linkedin.com/in/yonycalsin"
        rel="noreferrer"
        target="_blank"
      >
        Linkedin
      </a>
      <a
        class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
        href="https://twitter.com/yonycalsin"
        rel="noreferrer"
        target="_blank"
      >
        Twitter
      </a>
      <a
        class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
        href="mailto:helloyonycalsin@gmail.com"
        rel="noreferrer"
        target="_blank"
      >
        Email
      </a>
    </div>
  </div>
</body>
`;
